### PR TITLE
Derive Default for FluentBuilders

### LINKED ISSUE #363: FluentBuilders should be `Debug`
```rust
let req = client.list_tables();
// this should work
println!("{:?}", req);
req.send()
```

We should also consider what we actually want in this debug impl to be useful, but, as a baseline, a `Debug` that delegates to the `Debug` on the inner builder is probably sufficient.

### PR DESCRIPTION
**Pull request by declanvk**
*Issue #, if available:*
Closes #363 

*Description of changes:*
Update the `fluent_builder` codegen to derive `std::fmt::Default` on the structure.

I wasn't sure what the right place for testing this was, I could add some print outs to the [`aws/sdk/examples`](https://github.com/awslabs/smithy-rs/tree/main/aws/sdk/examples) crates.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by rcoh** [comment]
Can you add a test here: https://github.com/awslabs/smithy-rs/blob/main/aws/sdk/integration-tests/kms/tests/sensitive-it.rs? 
We've been collecting a bunch of trait-impl assertions there. Once you've added it there, you can run the command below to make sure everything compiles:
```
./gradlew :aws:sdk:cargoCheck
```


Could definitely also update one of the examples for a "realistic" need to debug the builder

And thanks for the PR!

File: aws/sdk/integration-tests/kms/tests/sensitive-it.rs [hunk 1]
```diff
@@ -54,6 +54,13 @@ fn client_is_clone() {
     let _ = client.clone();
 }
 
+#[test]
```
Comment (rcoh) [inline aws/sdk/integration-tests/kms/tests/sensitive-it.rs:57]: Can you use the pattern above to make this a compile failure instead of a test failure?

Comment (declanvk) [reply aws/sdk/integration-tests/kms/tests/sensitive-it.rs:57]: Sure, I'll copy the Send + Sync test.

**Review by rcoh — COMMENTED** [review]
LGTM, my brain was clearly not fully on when I said that this wouldn't be a compile failure

**Comment by declanvk** [comment]
> LGTM, my brain was clearly not fully on when I said that this wouldn't be a compile failure

I can still do it as a compile assertion on a type if you like?

**Comment by rcoh** [comment]
sure, might make the intention a little clearer if it fails in the future

**Review by rcoh — APPROVED** [review]

**Pull request merged**